### PR TITLE
Made @woocommerce/components/Stepper Typescript

### PR DESCRIPTION
**Pull request by rjchow**
- Converted @woocommerce/components/Stepper to Typescript as a test case for working on #8282 

### Detailed test instructions:

Nothing should visibly change
- Check that there are no changes in OBW where Stepper is used

### CODE REVIEW

**Comment by adrianduffell** [comment]
@rjchow Thanks for this, overall it's looking pretty good to me!

Just double checking whether `git mv` was used to change the file to `index.tsx`? It will help to preserve existing file history.

**Comment by rjchow** [comment]
> @rjchow Thanks for this, overall it's looking pretty good to me!
> 
> Just double checking whether `git mv` was used to change the file to `index.tsx`? It will help to preserve existing file history.

@adrianduffell unfortunately `git mv` doesn't work if contents of the file changes in the same commit, so the only way for that to work is to do `git mv` in one commit, then make the TS changes in a separate one, and not squash the commits when merging. Should I do that?

**Comment by adrianduffell** [comment]
> unfortunately git mv doesn't work if contents of the file changes in the same commit, so the only way for that to work is to do git mv in one commit, then make the TS changes in a separate one, and not squash the commits when merging. Should I do that?

Ah I see, thanks. Yeah I think it's worth doing even though it's a bit of a chore. Would you need to create separate PRs for each step?

**Comment by rjchow** [comment]
@adrianduffell I've changed it to two commits, it won't need separate PRs
It's now possible to follow the git history of the renamed Stepper.tsx file by using the `--follow` flag in git like this:

without --follow:
> git log packages/components/src/stepper/index.tsx
image


with --follow:
> git log --follow packages/components/src/stepper/index.tsx
image

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 - Update line-height of SelectControl label to avoid truncated descenders in some typefaces and zoom levels. #8186
 - Added Typescript type declarations to build for @woocommerce/components #8282
+- Made @woocommerce/components/Stepper a Typescript file #8286
```
Comment (chihsuan) [inline packages/components/CHANGELOG.md:5]: @rjchow I think we should add a `.` to have a consistent style.

```suggestion
- Made @woocommerce/components/Stepper a Typescript file. #8286
```

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 - Update line-height of SelectControl label to avoid truncated descenders in some typefaces and zoom levels. #8186
 - Added Typescript type declarations to build for @woocommerce/components #8282
```
Comment (chihsuan) [inline packages/components/CHANGELOG.md:4]: We should change this, too.
```suggestion
- Added Typescript type declarations to build for @woocommerce/components. #8282
```

**Review by chihsuan — APPROVED** [review]
Nice work! Only left two minor comments. Pre-approving! 👍

**Comment by adrianduffell** [comment]
> @adrianduffell I've changed it to two commits, it won't need separate PRs It's now possible to follow the git history of the renamed Stepper.tsx file by using the `--follow` flag in git like this:

@rjchow Looks good! Thanks for figuring this out. I use GitLens and am seeing the file history properly there too.

Screen Shot 2022-02-16 at 11 34 03 am

I wonder if you might need to "Rebase and merge" this PR to preserve the 2 commits instead of squashing them?

**Pull request merged**